hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/table/catalog/HoodieHiveCatalog.java

```diff
@@ -382,6 +383,8 @@ private Table translateSparkTable2Flink(ObjectPath tablePath, Table hiveTable) {
         parameters.putAll(TableOptionProperties.translateSparkTableProperties2Flink(hiveTable));
         String path = hiveTable.getSd().getLocation();
         parameters.put(PATH.key(), path);
+        Map<String, String> hoodieProps = loadFromHoodiePropertieFile(path, hiveConf);
+        parameters.putAll(TableOptionProperties.translateSparkTableProperties2Flink(hoodieProps));
         if (!parameters.containsKey(FlinkOptions.HIVE_STYLE_PARTITIONING.key())) {
```

Comment:
Do we need to put the explicit table options here ? Which function needs this config and where we miss it then ?

Reply:
We can read the table config through `StreamerUtil.createMetaClient(xx).getTableConfig()`

Reply:
Yes, after Spark created the table, there are some table properties in .hoodie/hoodie.properties, such as hoodie.datasource.write.hive_style_partitioning. These attributes are not included in the hive table attribute.

Reply:
But we only append hive options here, curious why this can cause error ?

Reply:
Because spark uses the hoodie.datasource.write.hive_style_partitioning property to be true when creating hudi non-partitioned tables, and records hoodie.datasource.write.hive_style_partitioning=true in hoodie.properties. But this property is not seen in hoodiehivecatalog, and inferred It is also inferred that it is wrong, so hoodie.datasource.write.hive_style_partitioning is assigned a value of false and uploaded to hive options. As a result, spark has two different hoodie.datasource.write.hive_style_partitioning attribute values when viewing the table, thus reporting an error.

Reply:
if (!parameters.containsKey(FlinkOptions.HIVE_STYLE_PARTITIONING.key())) {
          Path hoodieTablePath = new Path(path);
          boolean hiveStyle = Arrays.stream(FSUtils.getFs(hoodieTablePath, hiveConf).listStatus(hoodieTablePath))
              .map(fileStatus -> fileStatus.getPath().getName())
              .filter(f -> !f.equals(".hoodie") && !f.equals("default"))
              .anyMatch(FilePathUtils::isHiveStylePartitioning);
          parameters.put(FlinkOptions.HIVE_STYLE_PARTITIONING.key(), String.valueOf(hiveStyle));
        }

--- The params in this code are not inferred according to the hoodie.properties property file, resulting in an inference error.

Reply:
Got you, i think we should not put extra options like `FlinkOptions.HIVE_STYLE_PARTITIONING` here, the right fix is to supplement the table config options for the flink read/write path, I have applied a patch actually.